pkg/kubectl/cmd/util/factory_builder.go

```diff
@@ -130,7 +130,7 @@ func (f *ring2Factory) PrintObject(cmd *cobra.Command, isLocal bool, mapper meta
 	if err != nil {
 		return err
 	}
-
+	mapping.GroupVersionKind = gvks[0]
 	printer, err := f.PrinterForMapping(cmd, isLocal, nil, mapping, false)
```

Comment:
@liggitt when use versionObject for printer. though my yaml is `v1beta2`, but mapping.GroupVersionKind  is `v1beta1`. add this line will fix this.
```
admindeMacBook-Pro-2:kubectl admin$ ./kubectl set image -f ../../examples/storage/minio/minio-distributed-statefulset.yaml *=test --local -o yaml
apiVersion: apps/v1beta2
kind: StatefulSet
metadata:
  creationTimestamp: null
  name: minio
spec:
  replicas: 4
...

```

Reply:
and last version kubectl, will update apiVersion to `v1beta1` though my yaml version is `v1beta2`:
```
admindeMacBook-Pro-2:kubernetes admin$ kubectl set image -f examples/storage/minio/minio-distributed-statefulset.yaml *=test --local -o yaml
apiVersion: apps/v1beta1
kind: StatefulSet
metadata:
  creationTimestamp: null
  name: minio
  namespace: default

```